### PR TITLE
Improve New TEG Experience

### PR DESCRIPTION
**Pull request by Azrun**
[help wanted][mapping][QoL][feature][balance]

## About the PR 
* Updates TEG instructions to describe circulator power usage and reagents. [**Quality of Life**]
  - [x] Pending review by community members.

* Update circulator blower to allow for player control. [**FEATURE**]
  * Allow for a LOT more consistent behavior especially with char burns.
  * Maximum pressure differential is adjusted to avoid people accidentally maxing it out and then immediately draining the APC cell.
image

* Update TEG description to provide a high level description of TEG variant and breadcrumbs to instructions. [**Quality of Life**]
image

* Adds TEG instructions to a couple maps. [**Quality of Life** and **MAPPING**]
* TEG instructions will be updated with an additional note if a RNG TEG is generated to describe the behavior. Example:
image

* Provides preliminary visual cue of circulator maintenance panel being open. [**Quality of Life**]
image

* Reduce amount of noise in TEG Serial Numbers. [**Quality of Life**]
  * Simplified scheme and reduce likelyhood of serial number collisions when new variants are added.

* ~~Provides Engineer Borg with Oil Can [**BALANCE**]~~
  * ~~Oil can has half-capacity and **CAN NOT** be used in chem dispensers.~~

* New Reagents [**FEATURE**]
  * Love and Hugs to degrump
  * Voltagen to drastically improve fan power consumption.  Use case is really for runaway temps.

## Why's this needed? 
Lower TEG barrier to entry by providing more information to the spessmans.
Provide visual cues to equipment state.
Reduce additional workload currently being put on Mentor and Admin staff from recent TEG changes.
Player control of target differential pressure will provide more control of engine output, specifically should allow for more stable outputs in char burns.

## Changelog

```
(u)Azrun:
(*)TEG instructions have been updated to capture latest changes.  Circulators description and instructions will provide an indication when the TEG is a special variant.  Circulators now have visual indication when maintenance hatch is open.  New less random serial number scheme.
(*)TEG target circulator blower pressure differential can be adjusted with a multitool. Just like the majority of other ATMOS pumps/gates!
(+)TEG circulators now support liquid electricity and hugs.  Show the circulators some TLC!
(+)TEG instructions added to a few maps.
```

### CODE REVIEW

**Comment by Kamades** [comment]
TEG Instructions:  Good!

RNG Notes - I honestly feel like this should be a second, blazingly obvious note.  Ideally on different coloured paper if that's possible.   Having it be a footnote to a note that most experienced engineers just ignore feels bad from a play design thing - whereas, putting a piece of paper on the circulator that is experimental (or both if both are experimental) is much more player facing.

Engineer Borg Oil Can - this... feels like the wrong way to do this.  This is also astonishingly powerful for borgs in terms of being able to speedboost themselves\other borgs - not bad per-se, but could end up feeling weird.  Also, putting non-oil in oil cans is going to be non-obvious for a lot of folks (oil cans are not exactly a go to reagent container).

Pure Love\Hugs - cute.
Voltagen - Uh, not going to say a lot here besides that encouraging enginerers to either raid chemistry or build their own chemistry feels pretty rough.  Asking sci nerds for secret chems tends to, well, get the scientist to have their back up out of fear you're spying on it, at least on RP.

**Comment by Kamades** [comment]
Also on the picture of the two sprites, I uh, literally don't see a difference.

**Comment by Awfullyfishy** [comment]
Very happy about these of these changes. TEG displaying the circulator power usage and reagents will significantly help people learn about the changes instead of it being hidden mechanics that they have to track down to find out about. The more information the better in my eyes. The maintenance panel visual change is pretty nice as well, just a big gaping hole is pretty noticeable. Are the effects of lube leaking or being drained noticeable as well? 

I'm not sure how to feel about the TEG instruction paper. The note at the bottom of a paper that nobody checks is a bit obscure. I'd prefer the TEG itself has a visual difference on its sprite, like a racing line or something, that prompts Engineers to check the TEG instruction paper to see what they're dealing with. Even if you told the entire player base to check the paper for RNG TEG details, I feel many would either not know about the papers existence or not remember to check when they notice their TEG is acting weird. I'd much prefer some more obvious hints that hint at the TEG being unique and lead the players to investigate.

I'm also not entirely sure how to feel about the oil can, I'm not entirely sure how useful or overpowered it might be, but it's worth a try? I'm all for more things for Cyborgs to do. Can you empty the can and fill it with say other lubricants to put in the TEG? I hope so. But it still feels a bit.. weird, since you can speedboost yourself and others I imagine. I don't mind testing it and reverting it if it's a bit too strong, but I feel like the vast majority of Engineering Cyborgs would ignore the usage for the TEG and just use it as a speedboost. Now if that's too overpowered or not, I can't say, but it's something at least.

And finally more reagents is always welcome. Lubricants and bad things seem like the very foundation for the system but the more reagents that are possible to be used the better. I want a lot more departments to work together and this is a great way to do so. My biggest fear is just Engineering making their own Chemistry setup and avoiding the fun part about this, but there's no real way to avoid that I suppose. Still maybe in the future you can have RNG TEGs that excel at using specific reagents to encourage people working together hopefully.

**Comment by Azrun** [comment]
I, likely stubbornly, enjoy the idea of someone wondering why something is different only to be asked if they checked the manual/instructions.  @ryebek brought up the idea of leveraging the engineer trait to surface things like “TEG like cold” in the description when examined and then reference the instructions so there is a path from A to B.

Oil
“Putting non-oil in oil cans is going to be non-obvious”, but isn’t that how all the reagent containers work though?  I assumed most borgs were more experienced players or they have bigger concerns such as “how do I equip things” instead of “how do I apply these reagents to the circulators to make the engine go brrr”.  @ryebek brought up having drag operations on containers have reagents transfer... but that would mean they can drag around an oil can and oil themselves up.
It does provide the droid more power but it is a limited resource if they use it on themselves, other droids, or TEG.

@Awfullyfishy
I do like the visual indication idea.  A periodic animation to show how beautiful and shiny it is might be neat too.  Great idea but I’d rather that be a separate PR.

Corporate has informed me that the pictures are identical.  @Kubius said he might be willing to take a stab at a proper sprite update but I’d prefer that be a separate PR to respect their time.  Currently it is just the open canister overlay transformed appropriately.

**Comment by nefarious6th** [comment]
I think one of my biggest problems with having better TEG guidance being trait-dependent is that very often on RP, it's not an engineer setting up the TEG. It's a ghost drone, a cyborg, a mime, a clown(gineer?), Captain, HoP, CWorker, NTSO; because engineering can be sparse no matter the time. New players shy away from it, so people who've worked with it before take it upon themselves to get something going because it's kind of an understood necessity for a 2-hour long RP round, regardless of if they're actually just a clown for this shift.

**Comment by Azrun** [comment]
I see your point but I'm not sure the clowngineer is going to be interested in the % improvements. They just want to keep the lights on. Additionally I counter with a terrible argument: Those are the characters who should RP wise be reading the instruction.  I was considering having additional details be engineering training or silicon.

**Comment by Azrun** [comment]
image
Gives spessmans the lever to adjust themselves.  Allows you to dial in needed pressure to maintain a stable char burn.

**Comment by Awfullyfishy** [comment]
While I don't mind it being clowngineers and HoPs not seeing it (as if they'd care about optimizing it anyway), Cyborgs lacking the ability to see it is a pretty bad problem. I'd rather it not be tied to a trait at all in that case purely so Cyborgs can see it.

**Comment by nefarious6th** [comment]
Just to leave a thought, I guess, on the engine stuff and the "thematics" and RP-wise stuff. Listen, I know you play primarily on RP. I usually do. Sometimes I wander over to main, so I see maybe a bit more of the odds and ends of cross-departmental/role experimentations and breaking things down to the essence of gameplay rather than storytelling. I think there's a real importance in knowing when to suspend a little reality on development changes and mechanics in order to ensure it's fun and approachable. I think restricting better engine feedback to a trait violates that. At the end of the day, I don't really look at a TEG and say "that's a department"; TEGs are mechanics to me. They're really, really critical ones. It's something that, to me, is greater than just the role of the entire engineering department or a subset of that, since power keeps the station going. Does it make RP sense for a clown to set up the engine on a round? Not really. I'm not really sure what you'd even say to make that skillset seem believable. But is it a courtesy to other players to set the TEG up regardless, because in a very meta-sense we know without an engine, power fails, and without power gameplay is absolutely miserable? Yes. I think there needs to be more consideration for walking that line in making these changes something mechanically accessible, even if it to you or me seems to violate a realistic RP narrative, because most rounds, a realistic RP narrative isn't the case, even when playing on the RP servers. Clowns and mimes will set up engines. I don't want to make that more difficult or impossible for the people who play them, or anything else.

**Comment by Azrun** [comment]
I observed a few minutes of an RP round to see how engines were doing… does that count? 

Thanks for your responses.  I think I understand where you are both coming from.  You both bring up valid points in how surfacing more is better. It can and probably could improve the general user experience. But it doesn't really jive with what I assume the intent of those training traits are, the intended audience of that feature, nor my design. This is intended to be a hint to RTFM. A visual cue on the sprite iff it occurs is going to have infinitely more value than a slightly different description text.

Is the +/- 15% more consequential than I understand it to be?  Will no one engage with the feature if non-engineers don't get the hint? Is there an expectation of playerbase confusion/backlash if they aren't getting the extra 150kW they were getting the day/round before? If they ask in discord then they are engaging with the community, hooray! If they are spamming ahelp/mentor then there is a problem.

Are these special variants supposed to be complete game changers that require completely different setups? I have plans for something like that but those are something people would seek out, antags may thrust upon you, (or admemes if they want 🤷) and not what you get at the start of a round.  Following the wiki should be enough.

If there are round start TEGs that require different setup instructions I think @Kamades suggestion of a separate bright colored paper with detailed instructions is the way forward.

If medical and engineering training traits are seen as problematic as a whole and act as a direct barrier to allowing the greytide to adequately run the station after the heads of staff have been disposed of, I think that is a conversation I'm interested in hearing/having.  To me they seem like opportunities to allow for different experiences to be had when performing different jobs besides equipment you spawn with.  I think the traits thematically tie into the tiered wage system, they get more things but more is expected of them.

As was pointed out to me borgs are metal frames. What is inside is just some squishy spessman grey matter. Maybe they were a captain, wizard, barman, janitor, engineer, or 🤡.  Should they get appropriate training as part of their module? I think that is a bigger question but for thematic reasons I lean more towards no personally (in this moment, maybe I'll have a different tune tomorrow).  Either way that is beyond the intended scope of the PR.

If these changes are contradictory to the vision of Goonstation or will put undue burden on staff im down to change it however is necessary.  Am I missing how big a deal this facet of this PR is? 😕

**Comment by nefarious6th** [comment]
I'm going to be a little blunt, because I really just don't even know how to give my thoughts on this anymore: you're asking a lot of questions in response to me, and I have a lot of thoughts, but truthfully? I don't have answers, because, throughout this entire process, I have never really gotten a solid idea on what the intent was. I get driving cross-departmental interactions and engine experiments, and whether that was done correctly is out of the scope of this particular PR. But the point of restricting information to fit some greater thematic vision? Making these mechanics more difficult to engage with for engineers, but also especially for non-engineers? I'm not going to lie: I don't really understand this angle at all. So when you ask about the 150kW, my answer is: I don't know. And some rounds 150kW is the difference between powering the PTL, and sometimes it's the difference between charging the second SMES or just the one, and sometimes it's the difference between whether the lights will stay on or not. And, for what it's worth, I think the people who've done the most R'ing of the F'ing M are probably the most lost on all this because we tried to understand these changes as they were coming in, while everyone else kept hellburning on past us and didn't bother to engage with these mechanics at all.

I can't even begin to engage your comments on "the greytide" and "metal frames", other than that I think you're beginning to perhaps miss out on the fact that those are players, who play this game, who interact with mechanics of the game, and the TEG is a mechanic of the game, and you seem particularly uninterested in the player experience part of what this system changes. I have really, really tried to give my thoughts on this but I feel like there's a fundamental difference in the way you see these changes and the way I have been experiencing them, and I'm not sure what else I can offer at this point.

**Comment by UrsulaMejor** [comment]
Don't limit instructional material to specific jobs through things like traits; let the player decide to roleplay that they don't see it if it'd make a better story.

I don't recommend giving oil can go engineer borgs but I'll defer to another developer if they think it's fine

**Comment by mordent-goonstation** [comment]
> I don't recommend giving oil can go engineer borgs

Agree. Now that the yarn berry tgui thing is in and there's only one tgui PR up, I'm back to working on some semblance of borg upgrades. The ability to *give* them an oil can seems fine, but not for them to start with one. For now, borgs can deal without this extra little optimization if they want to run the engine.

**Comment by Azrun** [comment]
Unfortunately this PR has 8 moving pieces all under a single umbrella so that isn't doing me any favors.  Me ramblings wasn't helping either.  I was trying to let you know the questions I was using to base my stance (even if is the wrong one).

The blower change should allow for consistent pre-Azrun output during char burns.  Was able to lock in ~2MW on Clarion that was dipping in power due to slow drop in furnace heat.  Loss of 15% will allow for 2 SMES charge at full with a lot to spare.  From my point of view if the SMES is charging at all then engineering is doing their job, poorly but they are doing it.  The PTL is there if you want to take it to the next level.

When implementing the description text I thought it was a neat spin mechanically and thematically.  Digging in more I thought it also matched the additional information medical training provides examining brains.  So to me the question started to shift to if this feature should have the check and should any feature.  It is still a bit of an apples and pears comparison to be honest.

There is a concern about people setting up a Chem Dispenser next to the TEG and bypassing Medical.  It also made since to me to have something that would be a road bump going the other way.  I want the departments to work together.  I think having differentiation between departments is a great way to get there, otherwise an army of staff assistants is just as good.

------

Sounds like I am the minority.  Thanks for talking thought it with me. I do really appreciate all of your time.

- [ ] Revert borg changes.

- [ ] Provide hints to all spessmans.

**Comment by Azrun** [comment]
> I get driving cross-departmental interactions and engine experiments, and whether that was done correctly is out of the scope of this particular PR. 

If anyone ever want to go into that with me or on the forums please do.  I might not _like_ what you have to say, but it is important that I hear.  You shouldn't have to feel like some random person is coming in and 💩 on your fun.

**Comment by UrsulaMejor** [comment]
how fast are chemical additives consumed? is 10u enough to last the round?

**Comment by Azrun** [comment]
Love and Hugs are consumed, everything else leverages standard reactions.  Lubricant doesn't degrade over time currently.  Exception to this if there is a leak which will currently random leak out  0.5 units. (Variables describing could be better named for code readability) 

- [ ] Improve naming and comments involved with consumption and loss of lubricant.

1 unit is enough to last the round.  I have plans to have a "low oil" state but that should come with providing additional player feedback of said state.

**Comment by Azrun** [comment]
Preliminary Wiki Updates

---

Use the multitool on the left and right circulators to toggle power to the Blower Manual override and set the target transfer pressure to 500. 

---

|- id="The left and right circulators?"
!...The left and right circulators?
| The circulators have two additional functions.
Lubrication system.  This setup does not require any special lubricant that some more advanced setups might benefit from. Still, if you're so curious, here's a general rundown:  The system can be accessed by using a screwdriver on the circulator and opening the access panel.  The viscosity of the fluids is used to determine how efficient the gas transfer from inlet to outlet are.  Additionally there are a number of [[chemicals]] that can directly impact how the circulator behaves.
Blower system.  Other setups may require more or less pressure to ensure steady power output.  Use of the blower does drain the APC so be careful with your experimentation to ensure you don’t drain the APC and battery.

**Comment by Azrun** [comment]
@ZeWaka requesting code review from yourself or some other poor soul.  Mordent is no longer _required_ as relevant change was reverted.

File: code/WorkInProgress/TempEngine.dm [hunk 1]
```diff
@@ -444,16 +546,28 @@
...
 				// to avoid confusion
 				src.variant_b = null
+				variant_description = null
 
 		src.circ1?.assign_variant(prepend_serial_num, src.variant_a, src.variant_b)
 		src.circ2?.assign_variant(prepend_serial_num, src.variant_a, src.variant_b)
 
+
+		SPAWN_DBG(0) // Offload this action...
+			for(var/obj/item/paper/engine/instructions in range(40,src))
```
Comment (ZeWaka) [inline code/WorkInProgress/TempEngine.dm:566]: 40? no.

Comment (ZeWaka) [reply code/WorkInProgress/TempEngine.dm:566]: better handled with a by_type imo

**Review by ZeWaka — APPROVED** [review]
code looks pretty good

File: code/WorkInProgress/TempEngine.dm (hunk 1 above)
Comment (Azrun) [reply code/WorkInProgress/TempEngine.dm:566]: Some CE offices are oddly distant from TEG, I'll try out by_type(). Honestly did not like my solution besides it being able to be offloaded by spawn at startup.

Comment (Azrun) [reply code/WorkInProgress/TempEngine.dm]: Added by_type tracking to TEG instructions.
Limited updating to current zlevel to account for prefab (that would have been an awkward bug to find)

Speeding up the search feels meh for what should occur once but round but it decouples the implementation from map design which is WAY better. Good call.

**Comment by Azrun** [comment]
@ZeWaka 
No known outstanding actions/issues.  Let me know if you want/need anything else.
Cheers🍻

**Review by pali6 — APPROVED** [review]
I think this looks good (except for the current merge conflict). If nothing else we can get a new round of feedback on the forum for what needs changing more.

**Pull request merged**